Transfer to ES6 modules.

TanVD opened the pull request:
From now on we can use AMD (requirejs loader) for app load instead of static includes right into html. It works faster, it is cleaner solution and it gives some sort of safety.

TanVD committed:
Add AMD modules.

Add requirejs support of amd modules. Also changes
d.ts for editor-core lib, few improvements for angular
(to make it work with requirejs) and TS (same).

TanVD committed:
Fix location of resources

Move directories with libs, css and other to the appropriate location.

TanVD committed:
Update angular.

Update angular from 1.1.5 version to 1.6.1. We are NOT upgrading to
angular 2.

TanVD committed:
Transfer to ES6.

Change grunt target to es6. Use Map from standart library instead of
own.

TanVD committed:
Fix share diagrams functionality.

Repair share button for diagrams. (Damaged by BPMN pull request).

TanVD committed:
Codacy fixes.

Fixes for Codacy styleguide.

jzuken commented:
"Subprograms" tab moved "Algorithms" line too much right:
![image](https://cloud.githubusercontent.com/assets/705678/23213958/10413c22-f91e-11e6-91cf-eda36c99853d.png)

jzuken commented:
"save as" window flashes and immediately disappears

jzuken commented:
is 2D model toolbar button gone? 
![image](https://cloud.githubusercontent.com/assets/705678/23214449/ce72b364-f91f-11e6-969b-16dd414ee54d.png)

jzuken commented:
robots link image is missing in the palette:
![image](https://cloud.githubusercontent.com/assets/705678/23214519/074204ec-f920-11e6-94aa-da59cdb7d961.png)

jzuken commented:
how can I add new points to links?

TanVD committed:
Fix various bugs

Fix 2dmodel button issue. Fix robots link image in the palette issue
(add link image from bpmn). Fix problems with css and images
downloading.

TanVD committed:
Fix images in bpmn.

SVGs in BPMN-service are damaged. For now fixed with convertation to
png. Also some fixes.

TanVD committed:
Fix BPMN saveAs

Fix saveAs problem in BPMN editor. Exactly as in a robots editor.

LChernigovskaya committed:
merge master

LChernigovskaya committed:
Merge branch 'create-modules' of https://github.com/qreal/wmp into create-modules

TanVD committed:
Palette icons fix

Fix palette icons in BPMN and Robots
